node.go

```diff
@@ -439,15 +458,12 @@ func (n *Node) handleVoteRequest(vreq *pb.VoteRequest) bool {
 
 	deny := &pb.VoteResponse{Term: n.term, Granted: false}
 
-	// Old term, reject
-	if vreq.Term < n.term {
+	// Old term or candidate's log is behind, reject
+	if vreq.Term < n.term || !n.handler.GrantVote(vreq.LogPosition) {
 		n.rpc.SendVoteResponse(vreq.Candidate, deny)
 		return false
 	}
 
-	// Save state flag
```

Comment:
Note that the test above is `vreq.Term < n.term` and `saveState` was previously set to `true` if `vreq.Term > n.term`, which means that if they were equal, we would not save the state. You are changing that. Is it intentional?